### PR TITLE
koord-scheduler: add elastic quota informer transformer

### PR DESCRIPTION
**Pull request by xulinfei1996**
### Ⅱ. Does this pull request fix one issue?
fixes https://github.com/koordinator-sh/koordinator/issues/1444

### V. Checklist

- [ ] I have written necessary docs and comments
- [ ] I have added necessary unit tests and integration tests
- [ ] All checks passed in `make test`

### CODE REVIEW

File: cmd/koord-scheduler/app/server.go [hunk 1]
```diff
@@ -350,6 +350,7 @@ func Setup(ctx context.Context, opts *options.Options, outOfTreeRegistryOptions
 	defaultprofile.AppendDefaultPlugins(cc.ComponentConfig.Profiles)
 
 	frameworkext.SetupCustomInformers(cc.InformerFactory)
+
```
Comment (eahydra) [inline cmd/koord-scheduler/app/server.go:353]: please roll back unrelated changes.

File: pkg/scheduler/plugins/elasticquota/elastic_quota_transformer.go [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/scheduler-plugins/pkg/apis/scheduling/v1alpha1"
+	"sigs.k8s.io/scheduler-plugins/pkg/generated/informers/externalversions"
+
+	apiext "github.com/koordinator-sh/koordinator/apis/extension"
+	"github.com/koordinator-sh/koordinator/pkg/util/transformer"
+)
+
+func setupElasticQuotaTransformers(factory externalversions.SharedInformerFactory) {
```
Comment (eahydra) [inline pkg/scheduler/plugins/elasticquota/elastic_quota_transformer.go:29]: recommend to move the file to `util/transformer`.

File: pkg/scheduler/plugins/elasticquota/elastic_quota_transformer.go [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+	if err != nil {
+		klog.Fatalf("Failed to create informer for resource %v, err: %v", resource.String(), err)
+	}
+	if err = informer.Informer().SetTransform(TransformElasticQuota); err != nil {
+		klog.Fatalf("Failed to SetTransform in informer, resource: %v, err: %v", resource, err)
+	}
+}
+
+var elasticQuotaTransformers = []func(elasticQuota *v1alpha1.ElasticQuota){
+	TransformElasticQuotaWithDeprecatedDeviceResources,
```
Comment (eahydra) [inline pkg/scheduler/plugins/elasticquota/elastic_quota_transformer.go:41]: Are existing ElasticQuota using deprecated Device resources? If you don't need it, or remove these functions first, but keep the overall transformer mechanism.

File: cmd/koord-scheduler/app/server.go (hunk 1 above)
Comment (xulinfei1996) [reply cmd/koord-scheduler/app/server.go:353]: fixed

File: pkg/scheduler/plugins/elasticquota/elastic_quota_transformer.go (hunk 3 above)
Comment (xulinfei1996) [reply pkg/scheduler/plugins/elasticquota/elastic_quota_transformer.go:41]: fixed

File: pkg/scheduler/plugins/elasticquota/elastic_quota_transformer.go (hunk 2 above)
Comment (xulinfei1996) [reply pkg/scheduler/plugins/elasticquota/elastic_quota_transformer.go:29]: fixed

**Review by eahydra — APPROVED** [review]
/lgtm
/approve

**Review by FillZpp — APPROVED** [review]
/lgtm
